compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.test.cpp

```diff
@@ -0,0 +1,401 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveUnnecessaryTransposeNetPass.h"
+
+#include <luci/IR/CircleNode.h>
+
+#include <luci/test/TestIOGraph.h>
+
+#include <gtest/gtest.h>
+
+namespace
+{
+
+using namespace luci::test;
+
+class TransposeReshapeTransposeGraph : public TestIOGraph
+{
+
+public:
+  void init(ShapeU32 in_shape, ShapeU32 front_perm, ShapeU32 mid_shape, ShapeU32 back_perm,
+            ShapeU32 out_shape)
+  {
+    TestIOGraph::init(in_shape, out_shape);
+
+    auto init_circle_const = [](luci::CircleConst *const_node, ShapeU32 shape) {
+      const_node->dtype(loco::DataType::S32);
+      const_node->size<loco::DataType::S32>(shape.size());
+      uint32_t i = 0;
+      for (auto v : shape)
+      {
+        const_node->at<loco::DataType::S32>(i++) = v;
+      }
+    };
+
+    _front_perm = g()->nodes()->create<luci::CircleConst>();
+    {
+      _front_perm->name("front_transpose/perm");
+      init_circle_const(_front_perm, front_perm);
+    }
+
+    _front_transpose = g()->nodes()->create<luci::CircleTranspose>();
+    {
+      _front_transpose->a(input());
+      _front_transpose->name("front_transpose");
+      _front_transpose->perm(_front_perm);
+    }
+
+    _mid_shape = g()->nodes()->create<luci::CircleConst>();
+    {
+      _mid_shape->name("mid_reshpae/shape");
+      init_circle_const(_mid_shape, mid_shape);
+    }
+
+    _mid_reshape = g()->nodes()->create<luci::CircleReshape>();
+    {
+      _mid_reshape->name("mid_reshape");
+      _mid_reshape->tensor(_front_transpose);
+      _mid_reshape->shape(_mid_shape);
+    }
+
+    _back_perm = g()->nodes()->create<luci::CircleConst>();
+    {
+      _back_perm->name("back_transpose/perm");
+      init_circle_const(_back_perm, back_perm);
+    }
+
+    _back_transpose = g()->nodes()->create<luci::CircleTranspose>();
+    {
+      _back_transpose->name("back_transpose");
+      _back_transpose->a(_mid_reshape);
+      _back_transpose->perm(_back_perm);
+    }
+
+    output()->from(_back_transpose);
+  }
+
+private:
+  luci::CircleTranspose *_front_transpose;
+  luci::CircleConst *_front_perm;
+
+  luci::CircleReshape *_mid_reshape;
+  luci::CircleConst *_mid_shape;
+
+  luci::CircleTranspose *_back_transpose;
+  luci::CircleConst *_back_perm;
+};
+
+} // namespace
+
+TEST(RemoveUnnecessaryTransposeNetPass, rank_reduction_pattern1)
+{
+  TransposeReshapeTransposeGraph g;
+  luci::RemoveUnnecessaryTransposeNetPass pass;
+
+  /**
+   * from efficient-former
+   *
+   * (1, 14, 14, 192)
+   *      A   B   C
+   *      |
+   * (1, 192, 14, 14)
+   *      C   A   B
+   *      |
+   * (1, 192, 196)
+   *      C   AB
+   *      |
+   * (1, 196, 192)
+   *      AB   C
+   */
+  g.init(/*in*/ {1, 14, 14, 192}, /*perm*/ {0, 3, 1, 2}, /*reshape*/ {1, 192, 196},
+         /*perm*/ {0, 2, 1}, /*out*/ {1, 196, 192});
+
+  EXPECT_TRUE(pass.run(g.g()));
```

Comment:
option)
if two `Transpose` are removed, graph would look like 
Input - Reshape - Output.
we can additionally check pass is working correctly if input of `Output` node is `Reshape`.

Reply:
The idea looks really good. 

If you don't mind, I'd like to add the additional check logic, later - After the pass and tests merged to the main, first.
( Because I'm a little afraid that I might be behind the planned schedule.)

Reply:
> ...  check logic, later

find with me :)